### DISCUSSION TITLE
Custom Babel Parser

### DISCUSSION THREAD

**Discussion by ngdriss** [babel/babel#16415] [Q&A]
Hello, I wanted to integrate a new requirement into the the parser which is variable as pointer (like in C/C++) where the syntax is: `function foo(&a, b){}` should produce an Identifier with pointer: true in the AST. I've been able to achieve this the only problem is that make build fails building the project and it is a bit hard to understand the reason of failure.

Is there a guide or example for this ?

Apperently it is failing at scripts/build-dist.js which is located in `babel-plugin-transform-runtime` package, is it possible to trigger the build only for babel parser ?

**Comment by liuxingbaoyu** [comment]
`yarn gulp build-babel` might be useful to you.
But I suspect the build fails because the change to the parser introduced a regression, which you may need to fix.

**Reply by ngdriss** [reply]
These are the changes i've made so far:

in line 679 of tokenizer/index.ts:
```
//check if next is a simple character then read variable identifier
    if (isIdentifierStart(next)) {
      this.eat(code);
      this.readWord(next);
      return;
    }
```
in line 2760 of expression.ts:
```
if (name[0] === "&") {
      node.pointer = true;
      node.name = name.slice(1);
    } else {
      node.name = name;
}
```
which was incorrect as I had to override the behavior inside `parseBindingAtom` at `lval.ts` file.

Thanks for your guidance 👍 now it is building and my new tests are working as expected.

**Comment by ngdriss** [comment]
@liuxingbaoyu is there a guide for making a release of the version I made and publish that package ? I'm really stuck at this point. Also yarn link doesn't work.

**Reply by nicolo-ribaudo** [reply]
`make prepublish-build` will build all the packages in the monorepo in production mode, then you can go in the package you want to publish and run `yarn pack` or `npm publish`. Or `make publish` if you want to publish all of them.
